src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,415 @@
+            Can be used as the setup name in the cinfdatabase
+        **kwargs (dict): Key-word arguments are passed to cinf Measurement.__init__
+            setup_name (str): The setup name in the database default to path_to_file
+            timestamp (str): Timestamp the measurement started given as
+                (YYYY-MM-DD HH:MM:SS)
+        """
+
+        if self.data_has_been_fetched:
+            print(
+                f"This {self.__class__.__name__} has already fetched data from "
+                f" {self.token} grouped by {self.grouping_column}"
+                " Returning the measurement resulting from the original read. "
+                "Use a new Reader if you want to read another file."
+            )
+
+            return self.measurement
+
+        self.measurement_class = kwargs.pop("measurement_class", cls)
+        self.setup_name = kwargs.pop("setup_name", path_to_file)
+        self.timestamp = kwargs.pop("timestamp", None)
+        self.comment = kwargs.pop("comment", None)
+        self.mass_scans = kwargs.pop("include_mass_scans", False)
+        self.verbose = kwargs.pop("verbose", False)
+        self.grouping_column = kwargs.pop("group", None)
+
+        # figure out whether to collect data group by 'comment' or 'timestamp'
+
+        if not self.grouping_column:
+            if self.timestamp and not self.comment:
+                self.grouping_column = "time"
+                self.token = self.timestamp
+            elif self.comment and not self.timestamp:
+                self.grouping_column = "comment"
+                self.token = self.comment
+            else:
+                warnings.warn(
+                    "Both a comment and a timestamp is given "
+                    "but no explicit grouping column is set. \n"
+                    f"Defaults to 'time' using '{self.timestamp}'",
+                    stacklevel=2,
+                )
+                self.grouping_column = "time"
+                self.token = self.timestamp
+
+        self.data_has_been_fetched = True
+
+        if issubclass(self.measurement_class, Spectrum):
+            return self.read_spectrums()
+
+        if issubclass(self.measurement_class, Measurement):
+            obj_as_dict = self.read_ms()
+            if issubclass(cls, self.measurement_class):
+                self.measurement_class = cls
+            obj_as_dict.update(kwargs)
+            self.measurement = self.measurement_class.from_dict(obj_as_dict)
+
+            if self.mass_scans:
+                if self.verbose:
+                    print("adding mass scans to the measurement")
+                self.add_mass_scans()
+
+            return self.measurement
+
+    def read_ms(self):
+        """Download MS data from cinfdata_database and make corresponding tseries and
+        vseries to place in a dictionary to return.
+        return obj_as_dict (dict)
+        """
+
+#        self.cinf_db = plugins.cinfdata.connect(
```

Comment:
I think these commented-out lines of code need to be deleted.

Reply:
Yes.